Feature: raise the maximum NewGRF limit to 255

TrueBrain opened the pull request:
## Motivation / Problem

Every limit is one limit too many! WE WANT MORE NEWGRFS! MOAR! MMOOAAARRRR!


## Description

As the Game Info now travels via TCP, we can increase the MTU limit on both sides to allow more NewGRFs. So .. I did exactly that.

I tested it with up to 209 NewGRFs .. that was all I had on disk :P

![image](https://user-images.githubusercontent.com/1663690/125176431-89bac580-e1d3-11eb-9083-b5f96379de7d.png)

Picture to proof it!

## Limitations

- 255 isn't enough. But .. a problem for another day ;)


## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

TrueBrain committed:
Feature: raise the maximum NewGRF limit to 255

James103 commented:
Why not just download at least 46 more NewGRFs from BaNaNaS and load them in addition to what you have already?

rubidium42 commented on src/newgrf.cpp:
```diff
@@ -80,7 +80,7 @@ static uint32 _ttdpatch_flags[8];
 
 static const uint MAX_SPRITEGROUP = UINT8_MAX; ///< Maximum GRF-local ID for a spritegroup.
-static const uint MAX_GRF_COUNT = 128; ///< Maximum number of NewGRF files that could be loaded.
+static const uint MAX_GRF_COUNT = 255; ///< Maximum number of NewGRF files that could be loaded.
```
The base set loads two NewGRFs, so practically you can now load only 253 NewGRFs in a game. Furthermore, it won't allow static NewGRFs to be loaded (for fonts or so) when the server has 253 or more NewGRFs set.

TrueBrain committed:
Fix a4987233: NewGRFs could no longer be loaded from the NewGRF folder.

This statement was removed by accident, as it felt it could be removed.
But it is used to know if the NewGRF is from the baseset folder or
from the NewGRF folder.

rubidium42 reviewed (approved):
Next rabbit hole please ;)
